server/etcdserver/api/v3rpc/watch.go

```diff
@@ -465,7 +466,10 @@ func (sws *serverWatchStream) sendLoop() {
 
 			// track id creation
 			wid := mvcc.WatchID(c.WatchId)
-			if c.Canceled {
+
+			verify.Assert((c.Canceled && c.Created) && wid != mvcc.InvalidWatchID, "the watch response invalid")
```

Comment:
```suggestion
			verify.Assert((c.Canceled && c.Created) && wid != mvcc.InvalidWatchID, fmt.Sprintf("unexpected watchId: %d, wanted: %d, since both 'Canceled' and 'Created' are true", wid, mvcc.InvalidWatchID))
```